core/pubnub_ccore_pubsub.c

```diff
@@ -504,6 +542,45 @@ enum pubnub_res pbcc_publish_prep(struct pbcc_context* pb,
                                 pb->subscribe_key);
     APPEND_URL_ENCODED_M(pb, channel);
     APPEND_URL_LITERAL_M(pb, "/0");
+
+#if PUBNUB_CRYPTO_API
+    if (NULL != pb->crypto_module) {
+        pubnub_bymebl_t message_block;
+        message_block.ptr = (uint8_t*)message;
+        message_block.size = strlen(message);
+
+        pubnub_bymebl_t encrypted = pb->crypto_module->encrypt(pb->crypto_module, message_block);
+
+        if (NULL == encrypted.ptr) {
+            PUBNUB_LOG_ERROR("pbcc_publish_prep(pbcc=%p) - encryption failed\n", pb);
+            free(message_block.ptr);
+            // TODO: add a specific error code for this
```

Comment:
any thing to do here because of TODO?

Reply:
I'll remove that - I found that we just return this error in case of encryption.